### ISSUE TITLE
building non-interactive

### ISSUE THREAD

**Issue by alexmyczko** [jpt/barlow#38]
how to build the font without interaction?

**Comment by jpt** [comment]
@alexmyczko can you be more specific about the kind of interaction you're talking about? if you mean building it from a shell terminal (i.e. outside a GUI), I don't believe it's currently possible. Right now Glyphsapp (or maybe also FontLab VI, which I believe can open Glyphs files) is the only way to build Barlow. It's possible in the future I could clean up the font sources so it would work with UFO format. This would involve, amongst other things, removing all overlaps of the glyph components, and creating a third master to consolidate all of the brace layers (a Glyphs feature not supported by the UFO standard).

**Comment by alexmyczko** [comment]
Yes I meant building it with a command, without GUI. To create packages of the font for Debian/Ubuntu...

UFO format, and/or sfd/.sfdir are both fine for me. That'd be great.

**Comment by jpt** [comment]
@alexmyczko Thanks for clarifying. I think it's unfortunate that an open source font like Barlow needs to rely on proprietary tools for the build step, and I'll keep this issue open, although to be honest it might be a long time before I can work on this.

IIRC, there are a couple of other fonts out there with programmatically generated intermediate masters to solve for the "brace layer" issue I'm describing, and I can look at them for inspiration. But the real issue is removing all overlaps to get the rounding of the letters correct, which will potentially increase file size, as well as create incompatible masters, which would mean a lot of work to keep both UFO and .glyphs files compatible. 

As an intermediate (and more realistic, given my own time constraints) step, would it satisfy your request if, instead of creating compatible UFO masters, I generated UFO files for each instance (all 54 styles)? That might not be as much work. The biggest "downside" I can think of is that it would rule out the possibility of a variable font (GX file). Although I'm not even sure if Debian supports variable fonts!

**Comment by davelab6** [comment]
Debian and gtk and gnome support Variable Fonts.

fontmake is on track to support compiling your glyphs file with brace and
bracket layers by April. Just be patient and do nothing :)

Barlow as a VF is on my wish list for next year.

**Comment by jpt** [comment]
> fontmake is on track to support compiling your glyphs file with brace and bracket layers by April. Just be patient and do nothing :)

perfect, I'm great at doing nothing :D 

does/will fontmake support `GlyphsFilterRoundCorner` for interior/exterior rounding as well? if not, it's easy enough to just apply the outer rounding manually before-hand, but it's that interior rounding where things get messy (for builds and ultimately for VF)

**Comment by taviks** [comment]
Any update on this? 

Would love to use Barlow as a variable font, but having to use a proprietary tool to build it is a bit of a no-go.

**Comment by jpt** [comment]
noninteractive static build is almost ready. VF will take more time. will update soon.
